smart-engine/src/main/java/org/smartdata/server/engine/rule/RuleExecutor.java

```diff
@@ -156,17 +156,22 @@ public String genVirtualAccessCountTable(List<Object> parameters) {
     String countFilter = "";
     List<String> tableNames =
         getAccessCountTablesDuringLast(interval);
-    return generateSQL(tableNames, newTable, countFilter);
+    return generateSQL(tableNames, newTable, countFilter, adapter);
   }
 
   @VisibleForTesting
-  static String  generateSQL(List<String> tableNames, String newTable, String countFilter) {
-    String sqlFinal;
+  static String  generateSQL(List<String> tableNames, String newTable, String countFilter, MetaStore adapter) {
```

Comment:
The call stack of this function is really complicated, hope someone can refactor it in the future :(